amqp/src/main/scala/akka/stream/contrib/amqp/model.scala

```diff
@@ -18,46 +19,129 @@ sealed trait AmqpSourceSettings extends AmqpConnectorSettings
   noLocal:            Boolean                = false,
   exclusive:          Boolean                = false,
   consumerTag:        String                 = "default",
   arguments:          Map[String, AnyRef]    = Map.empty
-) extends AmqpSourceSettings
+) extends AmqpSourceSettings {
+  @annotation.varargs
+  def withDeclarations(declarations: Declaration*) = copy(declarations = declarations.toList)
+
+  def withNoLocal(noLocal: Boolean) = copy(noLocal = noLocal)
+
+  def withExclusive(exclusive: Boolean) = copy(exclusive = exclusive)
+
+  def withConsumerTag(consumerTag: String) = copy(consumerTag = consumerTag)
+
+  def withArguments(argument: (String, AnyRef), arguments: (String, AnyRef)*) =
+    copy(arguments = (argument +: arguments).toMap)
+
+  @annotation.varargs
+  def withArguments(argument: akka.japi.Pair[String, AnyRef], arguments: akka.japi.Pair[String, AnyRef]*) =
+    copy(arguments = (argument +: arguments).map(_.toScala).toMap)
+}
+
+object NamedQueueSourceSettings {
+  /**
+   * Java API
+   */
+  def create(connectionSettings: AmqpConnectionSettings, queue: String) =
+    NamedQueueSourceSettings(connectionSettings, queue)
+}
 
 final case class TemporaryQueueSourceSettings(
   connectionSettings: AmqpConnectionSettings,
   exchange:           String,
-  declarations:       Seq[Declaration],
+  declarations:       Seq[Declaration]       = Seq.empty,
   routingKey:         Option[String]         = None
-) extends AmqpSourceSettings
+) extends AmqpSourceSettings {
+  def withRoutingKey(routingKey: String) = copy(routingKey = Some(routingKey))
+
+  @annotation.varargs
+  def withDeclarations(declarations: Declaration*) = copy(declarations = declarations.toList)
+}
+
+object TemporaryQueueSourceSettings {
+  /**
+   * Java API
+   */
+  def create(connectionSettings: AmqpConnectionSettings, exchange: String) =
+    TemporaryQueueSourceSettings(connectionSettings, exchange)
+}
 
 final case class AmqpSinkSettings(
   connectionSettings: AmqpConnectionSettings,
-  exchange:           Option[String],
-  routingKey:         Option[String],
-  declarations:       Seq[Declaration]
+  exchange:           Option[String]         = None,
+  routingKey:         Option[String]         = None,
+  declarations:       Seq[Declaration]       = Seq.empty
 ) extends AmqpConnectorSettings {
+  def withExchange(exchange: String) = copy(exchange = Some(exchange))
+
+  def withRoutingKey(routingKey: String) = copy(routingKey = Some(routingKey))
+
+  @annotation.varargs
+  def withDeclarations(declarations: Declaration*) = copy(declarations = declarations.toList)
+}
+
+object AmqpSinkSettings {
+  /**
+   * Java API
+   */
+  def create(connectionSettings: AmqpConnectionSettings) =
+    AmqpSinkSettings(connectionSettings)
 }
 
 /**
  * Only for internal implementations
  */
 sealed trait AmqpConnectionSettings
 
-case object DefaultAmqpConnection extends AmqpConnectionSettings
+case object DefaultAmqpConnection extends AmqpConnectionSettings {
```

Comment:
Doc: "Connects to a local AMQP broker at the default port with no password" or something like that.